master/internal/rm/kubernetesrm/job_test.go

```diff
@@ -0,0 +1,285 @@
+
+	return newPod, aID, sub
+}
+
+func setupEntrypoint(t *testing.T) {
+	err := etc.SetRootPath("../../../static/srv")
+	if err != nil {
+		t.Logf("Failed to set root directory")
+	}
+}
+
+func checkReceiveTermination(
+	t *testing.T,
+	update *batchV1.Job,
+	newJob *job,
+	sub *sproto.ResourcesSubscription,
+) {
+	state, err := newJob.jobUpdatedCallback(update)
+	switch {
+	case err != nil, state == cproto.Terminated:
+		newJob.finalize()
+	}
+	time.Sleep(time.Second)
+
+	require.True(t, waitForCondition(time.Second, func() bool {
+		return sub.Len() == 1
+	}), "didn't receive termination event in time")
+	message := sub.Get()
+	containerMsg, ok := message.(*sproto.ResourcesStateChanged)
+	if !ok {
+		t.Errorf(
+			"expected sproto.ResourcesStateChanged but received %s",
+			reflect.TypeOf(message),
+		)
+	}
+	if containerMsg.ResourcesStopped == nil {
+		t.Errorf("container stopped message not present (state=%s)", containerMsg.ResourcesState)
+	}
+
+	assert.Equal(t, newJob.container.State, cproto.Terminated)
+}
+
+func TestResourceCreationFailed(t *testing.T) {
+	setupEntrypoint(t)
+
+	const correctMsg = "already exists"
+
+	ref, aID, sub := createJobWithMockQueue(t, nil)
+
+	purge(aID, sub)
+	assert.Equal(t, sub.Len(), 0)
+	// Send a second start message to trigger an additional resource creation failure.
+	err := ref.start()
+	require.NoError(t, err)
+	time.Sleep(time.Second)
+
+	// We expect two messages in the queue because the pod actor sends itself a stop message.
+	require.True(t, waitForCondition(time.Second, func() bool {
+		return sub.Len() == 2
+	}), "didn't receive termination event in time")
+	message := sub.Get()
+	containerMsg, ok := message.(*sproto.ContainerLog)
+	if !ok {
+		t.Errorf("expected sproto.ContainerLog but received %s", reflect.TypeOf(message))
+	}
+	assert.ErrorContains(t, errors.New(*containerMsg.AuxMessage), correctMsg)
+}
+
+func TestReceivePodStatusUpdateTerminated(t *testing.T) {
+	setupEntrypoint(t)
+
+	t.Run("job deleting, but in pending state", func(t *testing.T) {
+		t.Logf("Testing PodPending status")
+		ref, aID, sub := createJobWithMockQueue(t, nil)
+		purge(aID, sub)
+		assert.Equal(t, sub.Len(), 0)
+
+		ref.jobDeletedCallback()
+		ref.finalize()
+
+		require.True(t, waitForCondition(time.Second, func() bool {
+			return sub.Len() == 1
+		}), "didn't receive termination event in time")
+		message := sub.Get()
+		containerMsg, ok := message.(*sproto.ResourcesStateChanged)
+		if !ok {
+			t.Errorf(
+				"expected sproto.ResourcesStateChanged but received %s",
+				reflect.TypeOf(message),
+			)
+		}
+		if containerMsg.ResourcesStopped == nil {
+			t.Errorf("container stopped message not present (state=%s)", containerMsg.ResourcesState)
+		}
+
+		assert.Equal(t, ref.container.State, cproto.Terminated)
+	})
+
+	t.Run("job failed", func(t *testing.T) {
+		t.Logf("Testing PodFailed status")
+		ref, aID, sub := createJobWithMockQueue(t, nil)
+		purge(aID, sub)
+		assert.Equal(t, sub.Len(), 0)
```

Comment:
require.Zero(t, sub.Len())

Reply:
chatted offline about require vs assert (or not using either)